rules/xcodeproj.bzl

```diff
@@ -263,13 +280,27 @@ def _xcodeproj_impl(ctx):
             "CLANG_ENABLE_MODULES": "YES",
             "CLANG_ENABLE_OBJC_ARC": "YES",
         }
+
+        # Just like framework, we need to have absoluate path to the hmap files
+        # so that Objc files can correctly import headers
+        header_search_paths = []
+        for hmap in target_info.hmap_paths.to_list():
+            if len(hmap) == 0:
+                continue
+            if hmap != "." and hmap[0] != "/":
+                hmap = "$BAZEL_WORKSPACE_ROOT/%s" % hmap
+            header_search_paths.append("\"%s\"" % hmap)
+        header_search_paths.append("\"$BAZEL_WORKSPACE_ROOT\"")
```

Comment:
For correctness (and could do this in a separate PR), we should read the appropriate values from the compilation context:

```
  /**
   * Returns the immutable list of include directories to be added with "-I" (possibly empty but
   * never null). This includes the include dirs from the transitive deps closure of the target.
   * This list does not contain duplicates. All fragments are either absolute or relative to the
   * exec root (see {@link
   * com.google.devtools.build.lib.analysis.BlazeDirectories#getExecRoot(String)}).
   */
  public ImmutableList<PathFragment> getIncludeDirs() {
    return commandLineCcCompilationContext.includeDirs;
  }

  /**
   * Returns the immutable list of include directories to be added with "-iquote" (possibly empty
   * but never null). This includes the include dirs from the transitive deps closure of the target.
   * This list does not contain duplicates. All fragments are either absolute or relative to the
   * exec root (see {@link
   * com.google.devtools.build.lib.analysis.BlazeDirectories#getExecRoot(String)}).
   */
  public ImmutableList<PathFragment> getQuoteIncludeDirs() {
    return commandLineCcCompilationContext.quoteIncludeDirs;
  }
  /**
   * Returns the immutable list of include directories to be added with "-isystem" (possibly empty
   * but never null). This includes the include dirs from the transitive deps closure of the target.
   * This list does not contain duplicates. All fragments are either absolute or relative to the
   * exec root (see {@link
   * com.google.devtools.build.lib.analysis.BlazeDirectories#getExecRoot(String)}).
   */
  public ImmutableList<PathFragment> getSystemIncludeDirs() {
    return commandLineCcCompilationContext.systemIncludeDirs;
  }

  /**
   * Returns the immutable set of declared include directories, relative to a "-I" or "-iquote"
   * directory" (possibly empty but never null).
   */
  public NestedSet<PathFragment> getLooseHdrsDirs() {
    return looseHdrsDirs;
  }
```
So this means you can access this info by calling `target[CcInfo].compilation_context.include_dirs, quote_include_dirs, system_include_dirs and loose_hdrs_dirs` in the same way I accessed the `framework_includes` property of CompilationContext.

(I searched CcCompilationContext.java in bazel source for references to -I and -i.)

There is probably a corresponding set of info for swift in the swift_library providers.

Reply:
Oddly Tulsi does not seem to access these properties, unless I'm grepping wrong. I'm not sure why, it seems like the right thing to do.

Reply:
yeah definitely a separate PR to explore this way of getting the dirs. Given existing way is one of the correct ways to achieve what we want